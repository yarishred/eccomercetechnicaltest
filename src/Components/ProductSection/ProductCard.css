.card_product-container{
    width: 200px;
    height: 350px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: 1px solid var(--border_color);
    margin: 1rem;
}

.card_product-container .card-image{
    width: 200px;
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.card-image img{
    width: 90%;
    height: 90%;
}


.card_product-container .card-description{
    width: 100%;
    height: 150px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    padding: 1.2rem;
    background-color: var(--grey_background);
}

.card-description h5{
    font-weight: lighter;
    font-size: 0.8rem;
    margin-bottom: 0.5rem;
}

.card-description h3{
    color: var(--optional2_color)
}

.card-description button{
    display: block;
    padding: 5px 12px;
    margin: 0.5rem 0;
    font-family: var(--main_font);
    cursor: pointer;
    outline: none;
    color: var(--light_background);
    background: var(--dark_background);
    border: none;
}

