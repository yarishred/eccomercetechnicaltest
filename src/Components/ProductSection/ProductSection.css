.product-container{
width: 100%;
height: 85.5vh;
display: flex;
justify-content: space-evenly;
align-items: center;
padding: 0 4rem ;
}

.product-image{
    width: auto;
    height: auto;
    background-color: tomato;
    display: flex;
    justify-content: center;
    align-items: center;
}

img{
    display: block;
    width: 100%;
    height: 100%;
    object-fit: cover;
}


.product-description{
    width: 400px;
    height: 450px;
    display: flex;
    flex-direction: column;
    padding: 2rem ;
}

.product-description p{
    padding: 1rem 0;
    text-align: justify;
}

.product-description form input {
    display: block;
    margin: 0.5rem 0;
}

.product-description form input[type=number]{
    width: 60px;
    height: 30px;
    text-align: center;
    outline: none;
    font-weight: bold;
    font-family: var(--main_font);
}

.product-description form input[type=submit]{
    background-color: var(--color_background);
    font-weight: bold;
}