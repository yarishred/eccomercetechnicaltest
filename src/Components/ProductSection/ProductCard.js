import React from 'react'

import "./ProductCard.css"

export const ProductCard = ({title, price, thumbnail, currency_id}) => {
    return (
        <div className='card_product-container'>
            <div className='card-image' >
                <img src={thumbnail} alt={title} />
            </div>
            <div className='card-description'>
                <h5>{title}</h5>
                <h3>{`$ ${price} ${currency_id}`}</h3>
                <button>Agregar al Carrito</button>
            </div>
        </div>
    )
}
