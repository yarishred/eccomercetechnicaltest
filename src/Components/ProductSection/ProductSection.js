import React from 'react'

import "./ProductSection.css"

export const ProductSection = ({product}) => {

    const { title, price, thumbnail } = product

    return (
        <div className='product-container'>
            <div className="product-image">
                <img src={thumbnail} alt="" />
            </div>
            <div className="product-description">
                <h2>{title}</h2>
                <p>{title}</p>
                <h3>{`$ ${price} COP`}</h3>
                <br />
                <h4>Quantity</h4>
                <form action="">
                    <input type="number" defaultValue={1} />
                    <input type="submit" value="Add To Cart" />
                </form>
            </div>
        </div>
    )
}
