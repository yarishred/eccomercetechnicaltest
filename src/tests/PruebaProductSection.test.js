import React from "react";
import { shallow } from "enzyme";

import { ProductSection } from "../Components/ProductSection/ProductSection";

describe("Pruebas en <ProductSection/>", () => {
  test("Debe de mostrar <ProductSection/> correctamente", () => {
    const product = {
      currency_id: "COP",
      id: "MCO568321632",
      price: 20000,
      thumbnail:
        "http://http2.mlstatic.com/D_840425-MCO46714060635_072021-O.jpg",
      title: "Camibuso Cuello Redondo T-shirt Colores Manga Larga",
    };

    const wrapper = shallow(<ProductSection product={product} />);

    expect(wrapper).toMatchSnapshot();
  });
});
