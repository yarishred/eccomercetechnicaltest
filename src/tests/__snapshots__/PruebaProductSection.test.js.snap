// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductSection/> Debe de mostrar <ProductSection/> correctamente 1`] = `
<div
  className="product-container"
>
  <div
    className="product-image"
  >
    <img
      alt=""
      src="http://http2.mlstatic.com/D_840425-MCO46714060635_072021-O.jpg"
    />
  </div>
  <div
    className="product-description"
  >
    <h2>
      Camibuso Cuello Redondo T-shirt Colores Manga Larga
    </h2>
    <p>
      Camibuso Cuello Redondo T-shirt Colores Manga Larga
    </p>
    <h3>
      $ 20000 COP
    </h3>
    <br />
    <h4>
      Quantity
    </h4>
    <form
      action=""
    >
      <input
        defaultValue={1}
        type="number"
      />
      <input
        type="submit"
        value="Add To Cart"
      />
    </form>
  </div>
</div>
`;
